### ISSUE TITLE
Publish modern ESNext/ES2022 code

### ISSUE THREAD

**Issue by arv** [hoeck/simple-runtypes#86]
Right now the code is compiled to ES5 which is a lot more verbose than needed.

Can the esm.js file at least be compiled to ES6 if not esnext?

**Comment by hoeck** [comment]
Of course! Atm. the build and packaging is completely done with tsdx but that has not seen an update in years.

Any suggestions on how to replace tsdx with something maintained and more modern? I'd prefer not to build everything by hand.
Or could you point me to an existing repo that builds and publishes a TS library in all required formats?

**Comment by arv** [comment]
I've used esbuild for most of my needs in the past but it does not support generating `.d.ts` files.

vite has [plugins for `.d.ts`](https://github.com/vitejs/vite/discussions/7509) files and that looks promising to me but under the hood it just uses rollup + tsc for the bundled `.d.ts` so you might as well simply use tsc directly to generate the `.d.ts`  files
